File: src/05_functions.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import math
from datetime import date
from datetime import datetime
from dateutil.relativedelta import relativedelta

# ----------------
# String Functions
# ----------------
wordarray = ("Hello", "World")
words = "Hello World"

print("CAT:", "".join(wordarray))
print("CATX:", ",".join(wordarray))
print("CHAR:", words[0:1])
print("COMPBL:", " ".join("Hello World  with   lots    of     space".split()))
print("COMPRESS:", words.replace(" ", ""))
print("FIND:", words.find("World"))
print("INDEX:", words.find("World"))
print("INPUT:", int("42"))
print("LEFT:", "  Hello World  ".lstrip())
print("LENGTH:", len(words))
print("LOWCASE:", words.lower())
print("PUT:", str(42.0))
print("QUOTE:", '"{}"'.format(words))
print("REPEAT:", "z"*42)
print("SCAN:", words.split(" ")[1])
print("STRIP:", "  Hello World  ".strip())
print("TRANSLATE:", words.replace("l", "?"))
print("TRANWRD:", words.replace("World", "Earth"))
print("TRIM:", "  Hello World  ".rstrip())
print("UPCASE:", words.upper())

# ----------------------
# Mathematical Functions
# ----------------------

# ----------------------
# Mathematical Constants
# ----------------------
print("E:", math.e)
print("PI:", math.pi)

# ------------------
# Datetime Functions
# ------------------
d = date.today()
print("DATE: ", d)
dt = datetime.now();
print("DATETIME: ", dt)
t = datetime.time(datetime.now());
print("TIME: ", t)

print("DATEPART: ", datetime.date(dt))
print("DAY: ", d.day)
print("HOUR: ", t.hour)
print("INPUT: ", datetime.strptime('2017-02-28', '%Y-%m-%d'))
d1 = date(2016, 2, 12)
d2 = date(2017, 2, 6)
print("INTCK: ", relativedelta(d2, d1).months)
print("INTNX: ", d1 + relativedelta(months = 12))
print("MINUTE: ", t.minute)
print("MONTH: ", d.month)
print("PUT: ", str(d), "or", d.strftime('%d/%m/%Y'))
print("QTR: ", d.month//3+1)
print("SECOND: ", t.second)
print("TIMEPART: ", datetime.time(dt))
print("TODAY: ", date.today())
print("WEEK: ", d.isocalendar()[1])
print("WEEKDAY: ", d.weekday())
print("YEAR: ", d.year)
print("YRDIF: ", relativedelta(d2, d1).years)

# -----------------------
# Miscellaneous Functions
# -----------------------

# ----------------------
# User Defined Functions
# ----------------------

File: src/05_functions.r
library(lubridate)

setwd("~/sas-py-r/src")

# ----------------
# String Functions
# ----------------
wordarray <- c("Hello", "World")
words <- "Hello World"

print(paste("CAT:", paste(wordarray, collapse='') ))
print(paste("CATX:", paste(wordarray, collapse=',') ))
print(paste("CHAR:", substr(words, 0, 1) ))
print(paste("COMPBL:", gsub("\\s+", " ", "Hello World  with   lots    of     space") ))
print(paste("COMPRESS:", gsub(" ", "", words) ))
print(paste("FIND:", regexpr("World", words) ))
print(paste("IFC:", ifelse(1<2, "Correct", "Incorrect") ))
print(paste("INDEX:", regexpr("World", words) ))
print(paste("INPUT:", as.integer("42") ))
print(paste("LEFT:", trimws("  Hello World  ", "left") ))
print(paste("LENGTH:", nchar(words) ))
print(paste("LOWCASE:", tolower(words) ))
print(paste("PUT:", as.character(42) ))
print(paste("QUOTE:", paste('"', words, '"') ))
print(paste("REPEAT:", paste(rep('z', 42), collapse='') ))
print(paste("SCAN:", strsplit(words, " ")[[1]][2] ))
print(paste("STRIP:", trimws("  Hello World  ", "both") ))
print(paste("TRANSLATE:", gsub("l", "?", words) ))
print(paste("TRANWRD:", gsub("World", "Earth", words) ))
print(paste("TRIM:", trimws("  Hello World  ", "right") ))
print(paste("UPCASE:", toupper(words) ))

# ----------------------
# Mathematical Functions
# ----------------------

# ----------------------
# Mathematical Constants
# ----------------------
print(paste("E:", exp(1) ))
print(paste("PI:", pi ))

# ------------------
# Datetime Functions
# ------------------
d <- Sys.Date()
print(paste("DATE:", d ))
dt <- Sys.time()
print(paste("DATETIME:", dt ))
t <- dt - floor_date(dt, unit='days')
print(paste("TIME:", format(t) ))

print(paste("DATEPART:", date(dt) ))
print(paste("DAY:", day(d) ))
print(paste("HOUR:", hour(dt) ))
print(paste("INPUT:", as.Date("2017-02-28") ))
d1 <- ymd("2016-02-12")
d2 <- ymd("2017-02-06")
print(paste("INTCK:", (d1 %--% d2) %/% months(1) ))
print(paste("INTNX:", dt + years(1) ))
print(paste("MINUTE:", minute(dt) ))
print(paste("MONTH:", month(d) ))
print(paste("PUT:", format(d, "%d%b%Y") ))
print(paste("QTR:", quarters(d) ))
print(paste("SECOND:", second(dt) ))
print(paste("TIMEPART:", dt - floor_date(dt, unit='days') ))
print(paste("TODAY:", Sys.Date() ))
print(paste("WEEK:", week(d) ))
print(paste("WEEKDAY:", weekdays(d) ))
print(paste("YEAR:", year(d) ))
print(paste("YRDIF:", (d1 %--% d2) %/% years(1) ))

# -----------------------
# Miscellaneous Functions
# -----------------------

# ----------------------
# User Defined Functions
# ----------------------
mysum <- function(val1, val2) {
  return(val1 + val2)
}

print(paste("MYSUM: ", mysum(18, 24)))
